File: src/main/java/net/igneo/icv/networking/packet/EnchantHitS2CPacket.java
package net.igneo.icv.networking.packet;

import net.igneo.icv.init.ICVUtils;
import net.minecraft.network.FriendlyByteBuf;
import net.minecraftforge.network.NetworkEvent;

import java.util.function.Supplier;


public class EnchantHitS2CPacket {
    public EnchantHitS2CPacket() {
    
    }
    
    public EnchantHitS2CPacket(FriendlyByteBuf buf) {
    
    }
    
    public void toBytes(FriendlyByteBuf buf) {
    
    }
    
    public boolean handle(Supplier<NetworkEvent.Context> supplier) {
        NetworkEvent.Context context = supplier.get();
        context.enqueueWork(ICVUtils::clientCooldownDamageBonuses);
        return true;
    }
}

File: src/main/java/net/igneo/icv/networking/packet/EnchantHitS2CPacket.kt
package net.igneo.icv.networking.packet

import net.igneo.icv.init.ICVUtils
import net.minecraft.network.FriendlyByteBuf
import net.minecraftforge.network.NetworkEvent
import java.util.function.Supplier

class EnchantHitS2CPacket {

    constructor()

    constructor(buf: FriendlyByteBuf)

    fun toBytes(buf: FriendlyByteBuf) {
        // Empty implementation - no data to serialize
    }

    fun handle(supplier: Supplier<NetworkEvent.Context?>?): Boolean {
        val context = supplier!!.get()
        context?.enqueueWork { ICVUtils.clientCooldownDamageBonuses() }
        return true
    }
}